sklearn/mixture/_gaussian_mixture.py

```diff
@@ -42,7 +42,7 @@ def _check_weights(weights, n_components):
         )
 
     # check normalization
-    if not np.allclose(np.abs(1.0 - np.sum(weights)), 0.0):
+    if not np.allclose(np.abs(1.0 - np.sum(weights)), 0.0, atol=1e-6):
```

Comment:
Should we use a value here that depends on the dtype? Or will it always be the same dtype? I can't quite work it out in my head but maybe you know.

Reply:
Let me do that quickly.